source/postprocess/tracers.cc

```diff
@@ -93,8 +93,12 @@ namespace aspect
           last_output_time = this->get_time() - output_interval;
         }
 
-      // Advance the particles in the world to the current time
-      world.advance_timestep();
+      // Do not advect the particles in the initial refinement stage
+      const bool in_initial_refinement = (this->get_timestep_number() == 0)
+          && (this->get_pre_refinement_step() < this->get_parameters().initial_adaptive_refinement);
+      if (!in_initial_refinement)
+        // Advance the particles in the world to the current time
+        world.advance_timestep();
```

Comment:
Maybe I formulated that in a misleading way: The *advection* of the particles now only happens after the last refinement step so everything is correct with that (so this PR fulfils its purpose). 
During coding I noticed this peculiarity: The *output* of the first timestep is only written after the first refinement step (the first call to execute() will call the `output->output_particle_data(...)` function). After the next refinement when we call execute() the `last_output_time` will have been updated and therefore we will return early from this function (line 107). I think this behaviour is a bit strange, but it is the same in the visualization plugin, so I left it in its current state.
I do not know exactly how to explain the output change best, maybe a sketch helps. Lets assume you have 3 initial adaptive refinements, then with 'Run postprocessors on initial refinement = false' the output will work like this:
1. refinement.
2. refinement.
3. refinement <---- write output files (only call of the postprocessors)

With 'Run postprocessors on initial refinement = true' the output will work like this:

1. refinement <---- write output files (first call of postprocessrs)
2. refinement <---- output is skipped (early return from function)
3. refinement <---- output is skipped (early return from function)

If we want to change this we should change it in both the visualization and tracers postprocessor to stay consistent, but I am not sure if we should, and it is not related to this PR.